File: 3447-clear-digits/clear-digits.cpp
class Solution {
public:
    string clearDigits(string s) {
        int i = 0, j = 0;
        while (i < s.size()) {
            if (isdigit(s[i]))  j -= 1;
            else {
                s[j] = s[i];
                j++;
            }
            i++;
        }
        return s.substr(0,j);
    }
};

File: 3447-clear-digits/clear-digits.java
class Solution {
    public String clearDigits(String s) {
        char[] chars = s.toCharArray();
        int j = 0;
        for (int i = 0; i < chars.length; i++) {
            if (Character.isDigit(chars[i])) {
                j -= 1;
            } else {
                chars[j] = chars[i];
                j++;
            }
        }
        return new String(chars, 0, j);
    }
}